controllers/storagecluster/onboarding_validation_keys_generator_job.go

```
@@ -0,0 +1,109 @@
+
+var _ resourceManager = &onboardingValidationKeysGeneratorJob{}
+
+func (o *onboardingValidationKeysGeneratorJob) ensureCreated(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) (reconcile.Result, error) {
+
+	if !instance.Spec.AllowRemoteStorageConsumers {
+		r.Log.Info("Spec.AllowRemoteStorageConsumers is disabled, skipping onboarding validation key generator job creation")
+		return reconcile.Result{}, nil
+	}
+
+	if res, err := o.createJob(r, instance); err != nil {
+		return reconcile.Result{}, err
+	} else if !res.IsZero() {
+		return res, nil
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (o *onboardingValidationKeysGeneratorJob) createJob(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) (reconcile.Result, error) {
+	var err error
+	r.Log.Info("Spec.AllowRemoteStorageConsumers is enabled. test ritesh Creating Onboarding validation key generator job")
+	if os.Getenv(onboardingValidationKeysGeneratorImage) == "" {
+		err = fmt.Errorf("OnboardingSecretGeneratorImage env var is not set")
+		r.Log.Error(err, "No value set for env variable")
+		return reconcile.Result{}, err
+	}
+	publicKeySecret := &corev1.Secret{}
+	publicKeySecret.Name = onboardingValidationPublicKeySecretName
+	publicKeySecret.Namespace = instance.Namespace
+	actualSecret := &corev1.Secret{}
+	// Creating the job only if public is not found
+	err = r.Client.Get(r.ctx, client.ObjectKeyFromObject(publicKeySecret), actualSecret)
+	if errors.IsNotFound(err) {
+		onboardingSecretGeneratorJob := &batchv1.Job{}
+		onboardingSecretGeneratorJob.Name = onboardingValidationKeysGeneratorJobName
+		onboardingSecretGeneratorJob.Namespace = instance.Namespace
+		onboardingSecretGeneratorJob.Spec = *onboardingJobSpec.DeepCopy()
+		err = r.Client.Create(r.ctx, onboardingSecretGeneratorJob)
+		if err != nil {
+			r.Log.Error(err, "failed to create onboarding validation key generator job")
+			return reconcile.Result{}, err
+		}
+
+	}
```

you are failing silently here if you fail to get the secret as there's no `err != nil`

---

i didn't get this point. What you are expecting here?